sdk/azidentity/client_certificate_credential_test.go

```diff
@@ -82,6 +87,44 @@ func TestClientCertificateCredential_GetTokenSuccess_withCertificateChain(t *tes
 	}
 }
 
+func TestClientCertificateCredential_GetTokenSuccess_withCertificateChain_mock(t *testing.T) {
+
+	validateReq := func(req *http.Request) bool {
+		body, err := ioutil.ReadAll(req.Body)
+		if err == nil {
+			bodystr := string(body)
+			kvps := strings.Split(bodystr, "&")
+			assertion := strings.Split(kvps[0], "=")
+			token, _ := jwt.Parse(assertion[1], nil)
```

Comment:
Does this line return an error as the second returned object? If so, it's probably best to check this error here

Reply:
yes, but in the end the test will fail either way, I believe.

Reply:
It will, but if there is an error on parsing, it will fail because of a nil pointer exception whereas this will help with diagnosability

Reply:
actually, in this case we return an error even though the jwt parse is successful because we are passing nil for the validation function. I think we could check for JWT being null here instead though.